src/main/kotlin/no/skatteetaten/aurora/gobo/graphql/auroraapimetadata/ConfigNamesBatchDataLoader.kt

```diff
@@ -0,0 +1,13 @@
+package no.skatteetaten.aurora.gobo.graphql.auroraapimetadata
+
+import no.skatteetaten.aurora.gobo.GoboDataLoader
+import no.skatteetaten.aurora.gobo.graphql.GoboGraphQLContext
+import no.skatteetaten.aurora.gobo.integration.boober.AuroraApiMetadataService
+import org.springframework.stereotype.Component
+
+@Component
+class ConfigNamesBatchDataLoader(private val service: AuroraApiMetadataService) : GoboDataLoader<AuroraApiMetadata, List<String>>() {
+    override suspend fun getByKeys(keys: Set<AuroraApiMetadata>, ctx: GoboGraphQLContext): Map<AuroraApiMetadata, List<String>> {
+        return keys.associateWith { service.getConfigNames().names }
```

Comment:
Det er nok bare Kotlin-syntaksen som forvirrer meg, men jeg forstår ikke helt hva som skjer her. Hvordan får man assosiert riktig nøkkel med riktig navn? Skal det ikke være noe `it` her inne?

Reply:
Det som skjer her er at den looper over listen, og for hvert innslag vil den returnere keyen som key i mapet, og verdien (service.getConfigNames() her) som value i mapet.

Reply:
Når det gjelder `it` akkurat her, så er denne loaderen litt rar. for den bruker ikke keyen, den bare henter alle names. Så i praksis er denne loaderen bare for å gå fra graphql verden til spring verden.

Reply:
Ja ok, da er jeg med, takk!